src/app/admin/modules/program-management/admin-programs/components/admin-program-list/admin-program-list.component.ts

```diff
@@ -19,32 +21,60 @@ export class AdminProgramListComponent implements OnInit {
   departments: Department[] = [];
   /** The total number of programs. */
   isLoadingResults = true;
-
+  isSchoolAdmin = this.authService.isSchoolAdmin;
   constructor(
     private readonly route: ActivatedRoute,
     private readonly router: Router,
     private readonly programService: ProgramService,
+    private readonly departmentService: DepartmentService,
+    @Inject(AUTH_SERVICE) private readonly authService: AuthService,
   ) { }
 
   ngOnInit() {
-    this.programService.getPrograms({offset: 0, limit: 100}).pipe(
-      map(data => {
-        this.isLoadingResults = false;
-        this.programs = data.results;
-        return this.programs;
-      }),
-      catchError((err) => {
-        this.isLoadingResults = false;
-        return observableOf([]);
-      })).subscribe(
-      data => {
-        this.programs = data;
+    console.log(this.isSchoolAdmin);
+    const department = this.authService.department;
+    let departments: Department[] = [];
+    let secondId;
+    if (this.isSchoolAdmin) {
+      this.departmentService.getDepartments({offset: 0, limit: 200}).pipe(
```

Comment:
Provide an API in backend to retrieve top-level departments instead of retrieving all departments.